xml/System.Net/HttpListener.xml

```diff
@@ -372,6 +372,8 @@ Use the [HttpCfg.exe](/windows/win32/http/httpcfg-exe) tool to add SSL certifica
  The asynchronous operation must be completed by calling the <xref:System.Net.HttpListener.EndGetContext%2A> method. Typically, the method is invoked by the `callback` delegate.  
   
  This method does not block while the operation completes. To get an incoming request and block until the operation completes, call the <xref:System.Net.HttpListener.GetContext%2A> method.  
+ > [!NOTE]
```

Comment:
This is general APM note - do other APIs using APM have same warning? Should they have it? Should we rather clarify APM docs?

Reply:
The general APM concept of an "async callback" states it must be called on another thread:
https://docs.microsoft.com/en-us/dotnet/api/system.asynccallback?view=netcore-3.1
"Use an AsyncCallback delegate to process the results of an asynchronous operation in a separate thread."

I'm thinking there are 3 cases then:
1. Just BeginGetContext deviates from APM spec
2. All Begin* APIs in HttpListener deviate this way
3. All APM components in .Net system libraries behave this way

I hope it's just case 1 and in that case I'd much rather prefer we fix the deviation as a bug than document it, but I also know that HttpListener is not really supported anymore and it would change its behavior in a meaningful way (e.g. may some callers today expect the callback to happen on the same thread under the circumstances where it would happen (the request is already complete).

If we decide not to fix it and it is case 2, then yea the note should be propagated to all HttpListener Begin* methods, or on the main documentation page for HttpListener. 

If it's case 3 however, then it would be a bug between implementation and specification, and yep, we'd have to change the APM specification/documentation (assuming we can't fix all implementations :)). I do hope it's not this case though because even before reading APM docs, I already intuitively expected an "AsyncCallback" to be ... async. After reading the APM docs I can say at least that seems to be the expectation for .Net too.

Reply:
It is option 3 to my best knowledge and the docs are just not clear. It is not bug in spec.